internal/store/somestore/store.go

```diff
@@ -0,0 +1,28 @@
+package somestore
+
+import (
+	"context"
+)
+
+type Store interface {
+	UpdateGaugeMetric(ctx context.Context, arg UpdateGaugeMetricParams) (bool, error)
+	FindGaugeMetric(ctx context.Context, arg FindGaugeMetricParams) (Metric, error)
+	FindCounterMetric(ctx context.Context, arg FindCounterMetricParams) (Metric, error)
+	UpdateCounterMetric(ctx context.Context, arg UpdateCounterMetricParams) (bool, error)
+	FindAllMetrics(ctx context.Context, arg FindAllMetricsParams) ([]Metric, error)
+}
+
+type Impl struct {
+	s MemStorage
+}
+
+var _ Store = (*Impl)(nil)
+
+func New() Impl {
```

Comment:
Можно рассмотреть возможность возврата указателя на  `Impl`,  это позволит избежать копирования, тем более что далее работа ведется с указателем.

```suggestion
func New() *Impl {
```